src/librustc_query_system/query/plumbing.rs

```diff
@@ -51,6 +51,7 @@ pub struct QueryState<CTX: QueryContext, C: QueryCache> {
 }
 
 impl<CTX: QueryContext, C: QueryCache> QueryState<CTX, C> {
+    #[inline]
```

Comment:
This function is generic, so does it matter?

Reply:
Yes, `#[inline]` is an inline hint, while regular generic functions are just available for inlining.